src/inmanta/execute/runtime.py

```diff
@@ -310,24 +347,31 @@ class DelayedResultVariable(ResultVariable[T]):
 
     def __init__(self, queue: "QueueScheduler", value: Optional[T] = None) -> None:
         ResultVariable.__init__(self, value)
-        self.promises: List[IPromise] = []
-        self.done_promises: Set[IPromise] = set()
+        self.promises: Optional[List[IPromise]] = []
+        self.done_promises: Optional[Set[IPromise]] = set()
         self.queued = False
         self.queues = queue
         if self.can_get():
             self.queue()
 
     def get_promise(self, provider: "Statement") -> ISetPromise[T]:
         promise: ISetPromise[T] = SetPromise(self, provider)
```

Comment:
Would it matter, that in case `self.promises is None` you always return the same dummy. 

(i.e. construct an no-op `ISetPromise` on the module level an always return that same, single instance)

Reply:
I'll try but I don't think so. I think this is a pretty rare path: the result variable is frozen before even the name pointing to it could be dereferenced.

Reply:
No measurable impact.